src/lib.rs

```diff
@@ -849,6 +853,7 @@ fn parse_mail_recursive(
         .unwrap_or_else(|| ParsedContentType::default_conditional(in_multipart_digest));
 
     let mut result = ParsedMail {
+        raw_bytes: &raw_data[..raw_data.len() - 2],
```

Comment:
I am unsure why, but this `- 2` is needed, otherwise additional `\r\n` goes into the slice and signature validation fails.
Also this should be modified to avoid panic.

@staktrace Maybe you are aware of some part of the standard which define which "\r\n" are considered to be part of the message body and which are not? https://datatracker.ietf.org/doc/html/rfc1847#section-2.1 does not define this precisely.

Reply:
According to https://datatracker.ietf.org/doc/html/rfc3156#section-5 CRLF immediately preceding the delimiter is not included in the signed data, while all the trailing whitespace that remains does.

This is what I observe with Thunderbird: `raw_data` ends with `\r\n\r\n`, but the last two bytes are not counted when calculating the signature. So let's leave removal of the last two bytes as is, and if it breaks with some OpenPGP implementation later, add a fix and a test.

Reply:
Will move this code dealing with CRLF to deltachat core.